### ISSUE TITLE
`Data Editor` Do not reset SQL expression to filter results after reconnection

### ISSUE THREAD

**Issue by hyperion-cs** [dbeaver/dbeaver#23432]
**Is your feature request related to a problem? Please describe.**
In `Data Editor` I use SQL expressions to filter the results (i.e. "data") in a table for later viewing and editing. Sometimes I have the `Data` window open for quite a long time (several days or more). During this time, of course, the connection to the given database can be terminated. Unfortunately, after each reconnect, the filter is reset and I have to write it again (the same one!).
I mean that, for example, such a filter:
<img width="622" alt="image" src="https://github.com/dbeaver/dbeaver/assets/14217598/0d79be19-b12b-42ac-b82d-0ccd4df7b413">

After reconnect, it resets:
<img width="834" alt="image" src="https://github.com/dbeaver/dbeaver/assets/14217598/78137978-9be0-4ef7-bca3-5ff2fe2de91f">

**Describe the solution you'd like**
The solution is simple: do not reset this filter. It is probably better to make this behavior optional (i.e. add a bool flag to `Preferences -> Editors -> Data Editor`).

**Describe alternatives you've considered**
Type in the same filter every time after reconnecting 🤷🏻‍♂️

**Additional context**
Not required.

**Comment by E1izabeth** [comment]
Thank you for the request

**Comment by sashsvamir** [comment]
What happened to this feature? Did they finally add it?

**Comment by sashsvamir** [comment]
When I have multiple tabs database results and SQL expression entered everywhere, after I close/open the laptop, the connection has to be re-established and all the statements are reset. It's very inconvenient to use.

**Comment by E1izabeth** [comment]
Closing as a duplicate of #35536
If you're still interested in this request, please vote for the mentioned issue with 👍 under the issue description to promote it to a higher priority.

**Comment by hyperion-cs** [comment]
@E1izabeth, actually, chronologically speaking, #35536 is a duplicate of #23432, not the other way around.

**Comment by E1izabeth** [comment]
Yes, sure, you're right. However, the mentioned one already has several votes, so we decided to leave that one.
Thank you!
